### ISSUE TITLE
Wrong checksum calculation in handleUploadPart 

### ISSUE THREAD

**Issue by akshay-sawant-bh** [gaul/s3proxy#576]
if (contentMD5String != null) {
            try {
                contentMD5 = HashCode.fromBytes(
                        Base64.getDecoder().decode(contentMD5String));
            } catch (IllegalArgumentException iae) {
                throw new S3Exception(S3ErrorCode.INVALID_DIGEST, iae);
            }
            if (contentMD5.bits() != MD5.bits()) {
                throw new S3Exception(S3ErrorCode.INVALID_DIGEST);
            }
        }

In handleUploadPart the code only checks for 128 bits and not the actual value, so even if someone sends a wrong MD5 base64 encoded string of file which is 128 bits the code will not throw an error.

**Comment by akshay-sawant-bh** [comment]
This is for azureblob looks like we are not sending the actual content-MD5 received along with the Payload in case of blobstore

**Comment by gaul** [comment]
S3Proxy does not check the MD5 because this would require computing it on the entire blob before sending it to the storage backend.  Instead it takes the intended MD5 from the client and lets the storage backend enforce whether this is correct:

```java
        if (contentMD5 != null) {
            builder = builder.contentMD5(contentMD5);
        }
```

The jclouds `BlobMetadataToBlobProperties` code is hard to follow but I believe that it should send the Content-MD5 based on:

```java
      HttpUtils.copy(from.getContentMetadata(), to.getContentMetadata());
```

Can you provide a trace-level log that shows that this is not the case?

Also please consider testing the new `azureblob-sdk` which correctly sends the MD5.
